scripts/sandbox.js

```diff
@@ -1,40 +1,66 @@
-let sandboxLevel = "";
-let sandboxObject = ObstacleCube;
-let sanboxTpX = 3;
-let sandboxTpY = 3;
-let sandboxPortalColor = "Blue";
+const sandbox = {
+  history: [null],
+  level: "",
+  object: ObstacleCube,
+  tpX: 0,
+  tpY: 0,
+  portalColor: "Blue",
+};
 
 class Sandbox {
   static update() {
     UserInterface.displayText({
-      content: `x: ${Math.floor(mouseX / UNIT_SIZE)}, y:${Math.floor(
+      content: `x: ${Math.floor(mouseX / UNIT_SIZE)}, y: ${Math.floor(
         mouseY / UNIT_SIZE
       )}`,
       x: 1,
       y: 2,
     });
     this.drawMarking(mouseX, mouseY);
   }
+
+  static undo() {
+    const history = sandbox.history;
+
+    if (history.length - 1 > 0) {
+      history.splice(history.length - 1, 1);
+    }
+  }
+
   static pushLevel() {
+    const history = sandbox.history;
+
     scenes.splice(scenes.length - 1, 1, () => {
-      eval(sandboxLevel);
+      sandbox.level = history[history.length - 1];
```

Comment:
Why push history (line 53) and then immediately rewrite it with a function?

Why not just evaluate the history entry when it is time to run?